Improve PEcAn.SIPNET documentation and connect to PEcAn book (#3705)
PecanProject/pecan: models/sipnet/README.md

<!-- badges: start -->

[![Lifecycle: experimental](https://img.shields.io/badge/lifecycle-experimental-orange.svg)](https://lifecycle.r-lib.org/articles/stages.html#experimental)
[![PEcAn.SIPNET status badge](https://pecanproject.r-universe.dev/badges/PEcAn.SIPNET)](https://pecanproject.r-universe.dev/PEcAn.SIPNET)

<!-- badges: end -->

## What is SIPNET?

> SIPNET (Simplified Photosynthesis and Evapotranspiration Model) is a lightweight ecosystem model designed to simulate complex environmental dynamics. It tracks carbon, water, nitrogen, and trace greenhouse-gas (N₂O, CH₄) interactions, supporting applications from forest carbon inventories to managed croplands.

Reviewer: Realized we need to document both since PEcAn is still primarily used with v1. 

```suggestion
The SIPNET v1 (**S**implified PnET (**P**hotosynthesis** and **E**vapo**t**ranspiration)) model is a lightweight ecosystem model designed to simulate ecosystem carbon and water dynamics. 

SIPNET v2 added a nitrogen cycle, trace greenhouse-gas (N₂O, CH₄) fluxes, and support for cropland and ecosystem management. As of PEcAn v1.10, support for SIPNET v2 is partially implemented and under active development.
```

**Key Features:**
- Simulates photosynthesis, respiration, allocation, soil water, soil temperature, and nitrogen cycling
- Operates on sub-daily time steps
- Tracks vegetation, litter, soil, and mineral-N pools
- Event-based management for agricultural processes (planting, harvest, tillage, irrigation)
- Clean, modular, BSD-licensed codebase suitable for research and production

For comprehensive SIPNET documentation, see the [SIPNET project website](https://pecanproject.github.io/sipnet).

## Using SIPNET via PEcAn

PEcAn.SIPNET is the PEcAn interface package that enables SIPNET integration with the PEcAn workflow system. This package:
- Converts PEcAn-standard inputs to SIPNET format
- Manages SIPNET model configuration and execution

> - Processes SIPNET outputs back to PEcAn standards

Reviewer: ```suggestion
- Converts SIPNET outputs to netCDF in PEcAn standard format
```

> - Enables data assimilation, sensitivity analysis, and ensemble runs

Reviewer: ```suggestion
- Enables data assimilation, sensitivity analysis, and ensemble runs within the PEcAn modeling framework
```

**Getting Started:**

> - [Demo 1: Basic PEcAn Run](https://pecanproject.github.io/pecan-documentation/develop/basic-users-guide.html)

Reviewer: @AritraDey-Dev what is the correct URL for the new Demo 1?

Reviewer 2: It's here: https://pecanproject.github.io/pecan-documentation/develop/rendered-demo-notebooks/run_pecan.html.

SIPNET v2 added a nitrogen cycle, trace greenhouse-gas (N₂O, CH₄) fluxes, and support for cropland and ecosystem management. As of PEcAn v1.10, support for SIPNET v2 is partially implemented and under active development.

**Key Features:**
- Simulates photosynthesis, respiration, allocation, soil water, soil temperature, and nitrogen cycling
- Operates on sub-daily time steps
- Tracks vegetation, litter, soil, and mineral-N pools
- Event-based management for agricultural processes (planting, harvest, tillage, irrigation)
- Clean, modular, BSD-licensed codebase suitable for research and production

For comprehensive SIPNET documentation, see the [SIPNET project website](https://pecanproject.github.io/sipnet).

## Using SIPNET via PEcAn

PEcAn.SIPNET is the PEcAn interface package that enables SIPNET integration with the PEcAn workflow system. This package:
- Converts PEcAn-standard inputs to SIPNET format
- Manages SIPNET model configuration and execution
- Converts SIPNET outputs to netCDF in PEcAn standard format
- Enables data assimilation, sensitivity analysis, and ensemble runs

**Getting Started:**

> - [Demo 1: Basic PEcAn Run](https://pecanproject.github.io/pecan-documentation/develop/basic-users-guide.html)

Reviewer 2: ```suggestion
- [Demo 1: Basic PEcAn Run](https://pecanproject.github.io/pecan-documentation/develop/rendered-demo-notebooks/run_pecan.html)
```

install.packages('PEcAn.SIPNET')
```

Or install directly from GitHub:

```r
library(remotes)
install_github('pecanproject/pecan', subdir = "models/sipnet")
```

### Install SIPNET Model

To use PEcAn.SIPNET, you also need the SIPNET model executable installed on your system. For installation instructions, see:

> - [Installing SIPNET](https://pecanproject.github.io/pecan-documentation/develop/install-models.html#inst-sipnet)

Reviewer: This seems circular. Sorry to add more work but it seems that, if anything, that section of the page in the URL should link to the official sipnet install + this README (published as part of the package docs), but not duplicate the information.

> - [SIPNET Installation Guide](https://pecanproject.github.io/sipnet/install.html)

Reviewer 3: ```suggestion
To use PEcAn.SIPNET, you also need the SIPNET model executable installed on your system. You can download precompiled versions from the SIPNET [releases page](https://github.com/PecanProject/sipnet/releases), or [download the code](https://github.com/PecanProject/sipnet) from Github and compile it yourself following the official [instructions](https://github.com/PecanProject/sipnet/blob/master/docs/user-guide/getting-started.md).
```

## Documentation

- **PEcAn.SIPNET Package Docs:** https://pecanproject.github.io/package-documentation/develop/PEcAn.SIPNET/
- **SIPNET Model Docs:** https://pecanproject.github.io/sipnet
- **PEcAn Book - SIPNET Chapter:** https://pecanproject.github.io/pecan-documentation/develop/models-sipnet.html
- **Source Code:** https://github.com/PecanProject/pecan/tree/develop/models/sipnet

## Learn More

- **PEcAn Project:** https://pecanproject.github.io

> - **Tutorials:** https://pecanproject.github.io/tutorials.html

Reviewer 2: ```suggestion
- **Tutorials:** https://pecanproject.github.io/tutorials/
```
